week6/src/CreateTask.js

```diff
@@ -0,0 +1,131 @@
+  const [items, setItems] = useState(storedTask);
+  const [descriptions, setNewDescription]= useState("");
+  const [narration, setDescription] = useState(storedDescription);
+  const [time, setNewDate]= useState("");
+  const [date, setDate] = useState(storedDate);
+  useEffect(() => {
+    localStorage.setItem('items', JSON.stringify(items));
+    
+  }, [items]);
+  useEffect(() => {
+    localStorage.setItem('narration', JSON.stringify(narration));
+    
+  }, [narration]);
+  useEffect(() => {
+    localStorage.setItem('date', JSON.stringify(date));
+    
+  }, [date]);
+  function add(){
+    if (!tasks){
+      alert("please input task!");
+      return;
+    }
+    if (!descriptions){
+        alert("please input description!");
+        return;
+      }
+      if (!time){
+        alert("please input date!");
+        return;
+      }
+    const taskinput = /^(?=.*[a-z])/;
+    if (!taskinput.test(tasks)){
+      alert('task must contain at least one letter in');
+      return;
+    }
+    const descriptionInput = /^(?=.*[a-z])/;
+      if (!descriptionInput.test(descriptions)){
+        alert('description must contain at least one letter');
+        return;
+      }
+      const dateInput = /^(?=.*[0-9])/;
+      if (!dateInput.test(time)){
+        alert('date must contain at least one digit');
+        return;
+      }
+
+    const item = {
+      id: Math.floor(Math.random() * 1000),
+      value: tasks
+    };
+    
+    setItems(oldList => [...oldList, item]);
+    setNewitem(""); 
+  
+      const description = {
+        id: item.id,
+        value: descriptions
+      };
+      
+      setDescription(oldList => [...oldList, description]);
+      setNewDescription("");
+
+      const period = {
+        id: item.id,
+        value: time
+      };
+      
+      setDate(oldList => [...oldList, period]);
+      setNewDate("");
+      toggle();
+   }
+   function Delete(id){
+    const newArray1 = items.filter(item => item.id !== id);
+    const newArray2 = narration.filter(description => description.id !== id);
+    const newArray3 = date.filter(date => date.id !== id);
+    setItems(newArray1);
+    setDescription(newArray2);
+    setDate(newArray3);
+     } 
+  return (
+    <div>
+      <Modal isOpen={modal} toggle={toggle}>
+        <ModalHeader toggle={toggle}>Create Task</ModalHeader>
+        <ModalBody>
+          <form>
+          <div className='form-group'>
+          <label>Task Name</label>
+          <input placeholder='input task' type='text' className='form-control' value={tasks} onChange={e => setNewitem(e.target.value)}/>
+          </div>
+          <label>Description</label>
+          <textarea rows="5" className='form-control' value={descriptions} onChange={e => setNewDescription(e.target.value)}></textarea>
+          <label>Date created</label>
+          <textarea rows="2" className='form-control' value={time} onChange={e => setNewDate(e.target.value)}></textarea>
```

Comment:
date shouldn't be a plain textarea. we can use a date picker instead - https://www.w3schools.com/tags/att_input_type_date.asp

Reply:
that will help with properly validating the date too